src/test/scala/chiselTests/LazyCloneSpec.scala

```diff
@@ -57,6 +59,40 @@ class LazyCloneSpec extends ChiselFlatSpec {
       }))
     }
     ChiselStage.elaborate(new MyModule)
-    Counter.count should be(4L)
+    Counter.count should be(2L)
+  }
+
+  it should "clone Records because of external refs" in {
+    var count = 0L
+    class MyRecord(gen: UInt) extends Record with AutoCloneType {
+      lazy val elements = VectorMap("foo" -> gen)
+      count += 1
+    }
+    class MyModule extends RawModule {
+      val gen = UInt(8.W)
+      val in = IO(Input(new MyRecord(gen)))
+      val out = IO(Output(new MyRecord(gen)))
+      out := in
+    }
+    ChiselStage.elaborate(new MyModule)
+    count should be(4L)
+  }
+
+  it should "clone because of nested external ref" in {
+    var count = 0L
+    class MyBundle(gen: UInt) extends Bundle {
+      val foo = new Bundle {
+        val bar = gen
+      }
+      count += 1
+    }
+    class MyModule extends RawModule {
```

Comment:
Can we show the opposite case of not cloning here ?

Reply:
Done.